app/models/news/item.rb

```diff
@@ -42,6 +42,10 @@ def after_save
       to_remove.each(&method(:remove_collection))
     end
 
+    def before_destroy
+      collections.pluck(:id).each(&method(:remove_collection))
```

Comment:
Can we not use https://sequel.jeremyevans.net/rdoc-plugins/classes/Sequel/Plugins/AssociationDependencies.html?